src/test/java/com/kasperin/inventory_management/controllers/v1/ProcessedFoodControllerTest.java

```diff
@@ -0,0 +1,205 @@
+package com.kasperin.inventory_management.controllers.v1;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.kasperin.inventory_management.controllers.RestResponseEntityExceptionHandler;
+import com.kasperin.inventory_management.domain.FoodType;
+import com.kasperin.inventory_management.domain.ProcessedFood;
+import com.kasperin.inventory_management.services.ProcessedFoodService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+class ProcessedFoodControllerTest {
+
+    ObjectMapper objectMapper;
+
+    @Mock
+    ProcessedFoodService processedFoodService;
+
+    ProcessedFoodController controller;
+
+    MockMvc mockMvc;
+
+    @BeforeEach
+    void setUp() {
+        objectMapper = new ObjectMapper();
+        controller = new ProcessedFoodController(processedFoodService);
+
+        mockMvc = MockMvcBuilders.standaloneSetup(controller)
+                .setControllerAdvice(new RestResponseEntityExceptionHandler())
+                .build();
+    }
+
+    @Test
+    void findAll() throws Exception {
+        // given
+        ProcessedFood pf1 = new ProcessedFood();
+        pf1.setId(1L);
+        ProcessedFood pf2 = new ProcessedFood();
+        pf2.setId(2L);
```

Comment:
Note that I'm not adding a bunch of data. This test is just to exercise the logic, so we can just verify against the id attribute